### PR TITLE
Allow groupID to be specified when assigning partition

### PR DESCRIPTION
**Pull request by omarkj**
Motivation:

A Consumer Group can provide a lot of benefits even if the dynamic loadbalancing features are not used.

Modifications:

Allow for an optional GroupID when creating a partition consumer.

Result:

Consumer Groups can now be used when manual assignment is used.

### CODE REVIEW

**Comment by FranzBusch** [comment]
@swift-server-bot test this please

File: Sources/Kafka/Configuration/KafkaConsumerConfiguration.swift [hunk 1]
```diff
@@ -82,6 +82,23 @@ public struct KafkaConsumerConfiguration {
...
 
+        /// A consumption strategy based on partition assignment.
+        /// The consumer reads from a specific partition of a topic at a given offset.
+        ///
+        /// - Parameters:
+        ///     - partition: The partition of the topic to consume from.
+        ///     - group: The ID of the consumer group to commit to.
+        ///     - topic: The name of the Kafka topic.
+        ///     - offset: The offset to start consuming from. Defaults to the end of the Kafka partition queue (meaning wait for the next produced message).
+        public static func partition(
```
Comment (FranzBusch) [inline Sources/Kafka/Configuration/KafkaConsumerConfiguration.swift:93]: I think we can merge this with the above case. Can we also spell out in the `groupID` doc comment when one would want to provide it when assigning the consumer to a specific partition?

**Review by FranzBusch — APPROVED** [review]

**Comment by FranzBusch** [comment]
@swift-server-bot test this please

**Review by blindspotbounty — APPROVED** [review]
@omarkj thank you for the PR!
And sorry for the late review.. Could you rebase the branch to the latest main that CI would pass, please?

**Comment by blindspotbounty** [comment]
@swift-server-bot test this please

**Comment by FranzBusch** [comment]
@swift-server-bot test this please

**Comment by blindspotbounty** [comment]
@swift-server-bot test this please

**Comment by FranzBusch** [comment]
@swift-server-bot test this please

**Comment by FranzBusch** [comment]
@swift-server-bot test this please

**Pull request merged**